src/nickserv.c

```diff
@@ -708,7 +708,7 @@ static const char *
 visible_email_addr(struct userNode *user, struct handle_info *hi)
 {
     if (hi->email_addr) {
-        if (oper_has_access(user, nickserv, nickserv_conf.email_visible_level, 1)) {
+        if (hi == user->handle_info || oper_has_access(user, nickserv, nickserv_conf.email_visible_level, 1)) {
```

Comment:
> was to prevent information leakage in case someone's account gets compromised

If a user's account gets compromised that's a) on them and b) they have significantly more to worry about.

If somehow, some person guessed my authserv password, broke in to my email that they must already know to get the authcookie, used the `COOKIE` command, logged in, then by that point in time they already know my email so what use is it in hiding it? Most users have reasonably strict authmasks, so you must already know the email in the first place in order to get in.

I understand the rationale behind your reasoning, don't get me wrong. The initial reason I did this was because I lost access to a domain a few days ago and wanted to make sure that my authserv account wasn't tied to it. Rather than being able to check, I basically had to spam `/as checkemail` which seemed silly. On every other IRC network, I only had to `/ns info` so I had assumed this was just a bug.

> a lot of accounts I have normally only show a reacted email address

The only times I ever see this are in forgotten user name challenges, never after being singed in though.